src/replica/test/replica_test_base.h

```diff
@@ -68,7 +68,9 @@ struct replica_test_base : replica_stub_test_base
         mu->data.updates.emplace_back(mutation_update());
         mu->data.updates.back().code =
             RPC_COLD_BACKUP; // whatever code it is, but never be WRITE_EMPTY
-        mu->data.updates.back().data.assign(data.data(), 0, data.length());
+        std::shared_ptr<char> ptr(dsn::utils::make_shared_array<char>(data.length()));
+        memcpy(ptr.get(), data.data(), data.length());
+        mu->data.updates.back().data.assign(ptr, 0, data.length());
```

Comment:
```suggestion
        mu->data.updates.back().data = blob::create_from_bytes(std::string(data));
```

https://github.com/XiaoMi/rdsn/blob/master/include/dsn/utility/blob.h#L73